packages/golden_toolkit/README.md

```diff
@@ -265,7 +265,7 @@ import 'dart:async';
 
 import 'package:golden_toolkit/golden_toolkit.dart';
 
-Future<void> main(FutureOr<void> testMain()) async {
+Future<void> testExecutable(FutureOr<void> Function() testMain) async {
```

Comment:
based on doc https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html , the function should be called `main` I think 🤔

Reply:
I agree, I would prefer to keep this called `main` instead of `testExecutable` unless there is a specific functional reason for this change.

Reply:
In my case I was getting errors without the testExecutable function, so for me it was indeed a functional reason.
Here is the output I get with the function renamed to main:
```Running "flutter pub get" in flutter-bloc...                     1,355ms
/var/folders/bv/fqdwsfhd5xxgywh1_r9wzy3c0000gp/T/flutter_tools.2DkDgm/flutter_test_listener.OHcrSs/listener.dart:46:18: Error: Method not found: 'testExecutable'.
    return () => test_config.testExecutable(test.main);
                 ^^^^^^^^^^^^^^
Failed to load "/Users/nicolas/src/flutter-bloc/test/golden_test.dart":
Compilation failed
Test: /Users/nicolas/src/flutter-bloc/test/golden_test.dart
Shell: /opt/flutter/bin/cache/artifacts/engine/darwin-x64/flutter_tester
```

Reply:
@tsimbalar I had to struggle a lot bc I used that documentation at the beginning, but then i had to dig into the code and [it seeks for that specific name](https://github.com/flutter/flutter/blob/0a73ecf6df17038985eac12a9f5b0f7ef36f018c/packages/flutter_tools/lib/src/test/flutter_platform.dart#L204) and also [is used in the code that way too](https://github.com/flutter/flutter/blob/085f1daa430c794b4f9ae8e4bdeba8e4bfda2f81/packages/flutter_goldens/lib/flutter_goldens.dart#L30).

I don't know if it's a hack or what it actually is, I only know that that's the only way I have alltogether working ([golden test and this library with good pictures with fonts rendered](https://stackoverflow.com/questions/64788492/how-to-make-golden-tests-with-fonts-well-rendered-in-flutter))

Reply:
So weird. I have everything working on multiple repos with that function called `main` 🤔 

There we have : 
- Flutter v1.22.3

```
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.7 19H2, locale en-ES)
    • Flutter version 1.22.3 at /Users/yc00067/tools/flutter
    • Framework revision 8874f21e79 (3 weeks ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3
```


- `pubspec` with the following : 

```yaml
dev_dependencies:
  flutter_test:
    sdk: flutter

  golden_toolkit: ^0.8.0
```

Reply:
If this doesn't work, does it also mean that this documentation is inaccurate 🤔  ? 

https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html

Reply:
Looking a bit deeper it seems like it may have been a change introduced fairly recently 🤔 

https://github.com/flutter/flutter/pull/67425

Reply:
Ok, so result from digging is that as part of "Non nullable" work on `flutter`, `main` has been renamed to something else (`testExecutable`) because something in flutter null-checking enforces that `main` methods MUST return a `List` ... and therefore this `main` method does not comply and this gives issues.

My understanding is that this will really explode when people start using the "non null" previews etc ... At this point this is not really documented anywhere as far as I can see 🤔

Reply:
I understand that those changes will come in Flutter 1.24 https://github.com/flutter/flutter/releases/tag/1.24.0-7.0.pre

Reply:
Nice research! Yeah it seems this will break a lot of things... But well, we are ready for the future this way ;-)